content/posts/2022-08-17-ocp-on-m1.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "Running ocp on M1 machines"
+date: 2022-08-17T08:58:10+05:30
+---
+
+From the 2.4.1 release, CRC has been running on macOS aarch64 M1 CPUs.
+However, the support was limited to podman preset on M1. We've been asked numerous times when CRC openshift
+preset would get M1 support. This has changed with the recently released [crc 2.7.1](https://github.com/code-ready/crc/releases/tag/v2.7.1).
+This blog post will describe the challenges we had to overcome when we decided to add that support.
+
+# Background
+
+CRC is a tool which allows you to run an OpenShift cluster or podman containers on a laptop or local workstation.
+It can be used on a macOS, Windows or Linux machine.
+It runs its OpenShift cluster/podman runtime by starting a virtual machine (VM) using a disk image built in advance and packaged in what we call a _bundle_.
+On each platform, we try to use native hypervisors.
+
+# Issue of running OCP on M1
+
+Openshift container platform (OCP) uses Red Hat Core OS (RHCOS) which is based on latest RHEL-8 which is
+compiled for a 64K page size (aarch64), but Apple M1 processor uses 16K or 4K page sizes [link](https://access.redhat.com/discussions/5966451). 
```

Comment:
uses 4K or a maximum size of 16K,

Reply:
not sure if '(aarch64)' here is relevant, as I believe that is also the case for x86 IIRC.